proxy/router/src/lib.rs

```diff
@@ -29,21 +34,21 @@ pub trait Recognize {
     /// Errors produced by the discovered services
     type Error;
 
-    /// Key
+    /// Identifies a Route.
     type Key: Clone + Eq + Hash;
 
     /// Error produced by failed routing
     type RouteError;
 
-    /// The discovered `Service` instance.
+    /// A route.
     type Service: Service<Request = Self::Request,
                          Response = Self::Response,
                             Error = Self::Error>;
 
-    /// Obtains a Key for a request.
+    /// Determines a route to handle the given request.
```

Comment:
While overall I like your documentation changes here, I think this comment may have lost some information --- in particular, I think it should imply that this function returns a `Key` rather than a `Service`, and that the concrete service has to be acquired from `bind_service`...